lib/metric.js

```diff
@@ -0,0 +1,26 @@
+function Metric() {
+  this.count = 0;
+  this.time = 0;
+  this.startTime = undefined;
+}
+
+Metric.prototype.start = function() {
+  this.startTime = process.hrtime();
+  this.count++;
+};
+
+Metric.prototype.stop = function() {
+  var change = process.hrtime(this.startTime);
+
+  this.time += change[0] * 1e9 + change[1];
```

Comment:
This method will only work if there is a guarantee that we call stop before the next call to start. Looking at the implementation, I wouldn't think that we do have such a guarantee.

If we have this guarantee then you can ignore my comments on `startTime` above, if not we should refactor this to be something like the following.

```js
let startTime = metric.start();
...
metric.sop(startTime);
```